### PR TITLE
Added user config.yaml and handled user defined resource creation

### PR DESCRIPTION
**Pull request by akkumar29**

### CODE REVIEW

File: hooks/post_gen_project.py [hunk 1]
```diff
@@ -15,13 +16,34 @@ def main():
...
+
+    try:
+        if read_from_file is True:
+            with open(yaml_file, 'r') as stream:
+                output = yaml.safe_load(stream)
+                resources_name = output['resource_name']
+        else:
+            resources_name = resources_name.split(" ")
+    except FileNotFoundError as ex:
+        print("File not found")
```
Comment (Krishnkr) [inline hooks/post_gen_project.py]: @akkumar29  Can we add logging instead of print statement ? or May be we can follow other print statement

File: templates/{{ cookiecutter.project_slug }}/src/main/java/com/deloitte/controller/{{cookiecutter.resource_name}}Controller.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package com.deloitte.controllers;
```
Comment (Krishnkr) [inline templates/{{ cookiecutter.project_slug }}/src/main/java/com/deloitte/controller/{{cookiecutter.resource_name}}Controller.java:1]: Revert this changes for now, otherwise we need to to do import changes where ever is controller is used

Comment (akkumar29) [reply templates/{{ cookiecutter.project_slug }}/src/main/java/com/deloitte/controller/{{cookiecutter.resource_name}}Controller.java:1]: I have checked for implications and made the necessary changes!

File: hooks/post_gen_project.py (hunk 1 above)
Comment (akkumar29) [reply hooks/post_gen_project.py]: Done

**Review by Krishnkr — APPROVED** [review]
Great work @akkumar29
